module/testmodule/main.go

```diff
@@ -48,9 +48,13 @@ func mainWithArgs(ctx context.Context, args []string, logger logging.Logger) err
 		helperModel,
 		resource.Registration[resource.Resource, resource.NoNativeConfig]{
 			Constructor: newHelper,
-			Discover: func(ctx context.Context, logger logging.Logger) (interface{}, error) {
+			Discover: func(ctx context.Context, logger logging.Logger, extra map[string]interface{}) (interface{}, error) {
+				extraVal, ok := extra["extra"]
+				if !ok {
+					extraVal = "default"
+				}
 				return map[string]string{
-					"foo": "bar",
+					"extra": extraVal.(string),
```

Comment:
minor, I don't know what testmodule is doing, but it seems to me that this panics if extraVal is not a string

Reply:
Gotcha added type assertion